### PR TITLE
[DSM] PEPPER-139 

### PR DESCRIPTION
**Pull request by pegahtah**
## Context

_Say a few words about the bigger picture beyond this PR. How do your changes improve life for our users? Link directly to the jira ticket (just type "DDP-..." and it'll automatically link) to help your teammates quickly get up to speed on the motivation for these changes._

## Checklist

- [X] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/resources/liquibase/CMI/PEPPER-139_view_seq_order_status.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="PEPPER-139_view_seq_order_status" author="pegah">
+        <insert tableName="access_role">
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/resources/liquibase/CMI/PEPPER-139_view_seq_order_status.xml:9]: Is granting existing users view_seq_order_status role, a separate Data change ticket ? 
I mean inserting rows into access_user_role_group for the new role !

Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/resources/liquibase/CMI/PEPPER-139_view_seq_order_status.xml:9]: Yes, that's part of PEPPER-707

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/mercury/GetMercuryEligibleSamplesRoute.java [hunk 2]
```diff
@@ -38,7 +38,8 @@ protected Object processRequest(Request request, Response response, String userI
         }
         String realm = queryParams.get(RoutePath.REALM).value();
         String userIdRequest = UserUtil.getUserId(request);
-        if (!UserUtil.checkUserAccess(realm, userId, "kit_sequencing_order", userIdRequest)) {
+        if (!UserUtil.checkUserAccess(realm, userId, "kit_sequencing_order", userIdRequest) &&
+        !UserUtil.checkUserAccess(realm, userId, "view_seq_order_status", userIdRequest)) {
             log.warn("User doesn't have access " + userId);
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/mercury/GetMercuryEligibleSamplesRoute.java:43]: Just want to be sure you want an && here, not ||?

Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/mercury/GetMercuryEligibleSamplesRoute.java:43]: No, either of those accesses can see this page, so you should not have both in order to get the error

**Review by denniscunningham — APPROVED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/mercury/GetMercuryEligibleSamplesRoute.java (hunk 2 above)
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/mercury/GetMercuryEligibleSamplesRoute.java:43]: Could we express this in a positive way?
If the user has the role kit_sequencing_order or the user has view_seq_order_status, allow them to do 
    the action, 
else 
   deny access

**Review by otchet-broad — APPROVED** [review]

**Pull request merged**